cmd/evm/internal/t8ntool/execution.go

```diff
@@ -246,8 +245,10 @@ func (pre *Prestate) Apply(vmConfig vm.Config, chainConfig *params.ChainConfig,
 		if err != nil {
 			return nil, nil, nil, err
 		}
+		// TODO (rjl493456442) it's a bit weird to reset the tracer in the
+		// middle of block execution, please improve it somehow.
```

Comment:
So, the tracer can be set earlier, on the vmconfig before it is passed to the evm. The thing that changes between transactions is the `traceOutput`. So we would need to split up `getTracerFn`. I'll take a look

Reply:
Hm, yeah it's not quite that simple. Maybe let's leave it for a different PR